auto_report/release_stats.py

```diff
@@ -0,0 +1,183 @@
+"""Get release statistics from OpenSearch"""
+
+import collections
+from distutils.version import LooseVersion
+from opensearchpy import OpenSearch
+
+
+class ReleaseStats:  # pylint: disable=too-few-public-methods
+    """
+    Provides release stats from OpenSearch data.
+    """
+    def __init__(self, opensearch_client: OpenSearch, index: str, logger):
+        """Constructor, accepts opensearch client"""
+        self._os_client = opensearch_client
+        self._index = index
+        self._logger = logger
+
+    def get(self, from_date: str, to_date: str, top_n: int = None) -> dict:
+        """
+        Returns stats. Format:
+        {
+          "<release_tag>": {
+            "<install_result>": {
+              "value": "<absolute>", "percent": "<percent>
+            }
+          }
+        }
+        """
+        query = _get_query(from_date, to_date)
+        self._logger.debug("OpenSearch query: %s", query)
+        response = self._os_client.search(index=self._index, body=query)
+        self._logger.debug("OpenSearch response: %s", response)
+        stats = parse_response(response)
+
+        if not top_n:
+            return stats
+
+        return _get_top_by_version(stats, top_n)
+
+
+def _get_top_by_version(stats: dict, top_n: int) -> dict:
+    """
+    Returns top N releases, ordered by version
+    from latest to oldest
+    """
+
+    top_versions = sorted(stats, key=LooseVersion, reverse=True)[0:top_n]
```

Comment:
I will break once `2.10.0` will be released (on 29/09) because it's a lexical sort, right?

Reply:
Nope, it's version sort taken from `distutils.version` (`from distutils.version import LooseVersion`)
Loose because it has `v` prefix, AFAIU strict only works with only-number versioning